src/plugins/teleop/Teleop_TEST.cc

```diff
@@ -230,126 +191,42 @@ TEST_F(TeleopTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(KeyboardCommand))
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    QCoreApplication::processEvents();
-    sleep++;
-  }
+  igndbg << "Release A" << std::endl;
+  this->yawVel = 0.0;
+  this->KeyEvent(false, Qt::Key_A);
 
-  EXPECT_TRUE(received);
-  received = false;
-
-  // Generates a key press event on the main window.
-  QKeyEvent *keypress_A = new QKeyEvent(QKeyEvent::KeyPress,
-    Qt::Key_A, Qt::NoModifier);
-  app.sendEvent(win->QuickWindow(), keypress_A);
-
-  sleep = 0;
-  while (!received && sleep < maxSleep)
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    QCoreApplication::processEvents();
-    sleep++;
-  }
+  igndbg << "Release W" << std::endl;
+  this->forwardVel = 0.0;
+  this->KeyEvent(false, Qt::Key_W);
 
-  EXPECT_TRUE(received);
-  received = false;
+  igndbg << "Press X" << std::endl;
+  this->KeyEvent(true, Qt::Key_X);
 
-  // Generates a key release event on the main window.
-  QKeyEvent *keyrelease_A = new QKeyEvent(QKeyEvent::KeyRelease,
-    Qt::Key_A, Qt::NoModifier);
-  app.sendEvent(win->QuickWindow(), keyrelease_A);
+  igndbg << "Release X" << std::endl;
+  this->KeyEvent(false, Qt::Key_X);
 
-  sleep = 0;
-  while (!received && sleep < maxSleep)
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    QCoreApplication::processEvents();
-    sleep++;
-  }
-
-  EXPECT_TRUE(received);
-  received = false;
-  // Generates a key release event on the main window.
-  QKeyEvent *keyrelease_W = new QKeyEvent(QKeyEvent::KeyRelease,
-    Qt::Key_W, Qt::NoModifier);
-  app.sendEvent(win->QuickWindow(), keyrelease_W);
-
-  sleep = 0;
-  while (!received && sleep < maxSleep)
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    QCoreApplication::processEvents();
-    sleep++;
-  }
-
-  EXPECT_TRUE(received);
-  received = false;
-  // Generates a key press event on the main window.
-  QKeyEvent *keypress_X = new QKeyEvent(QKeyEvent::KeyPress,
-    Qt::Key_X, Qt::NoModifier);
-  app.sendEvent(win->QuickWindow(), keypress_X);
-
-  sleep = 0;
-  while (!received && sleep < maxSleep)
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    QCoreApplication::processEvents();
-    sleep++;
-  }
-
-  EXPECT_TRUE(received);
-  received = false;
-  // Generates a key release event on the main window.
-  QKeyEvent *keyrelease_X = new QKeyEvent(QKeyEvent::KeyRelease,
-    Qt::Key_X, Qt::NoModifier);
-  app.sendEvent(win->QuickWindow(), keyrelease_X);
-
-  sleep = 0;
-  while (!received && sleep < maxSleep)
-  {
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    QCoreApplication::processEvents();
-    sleep++;
-  }
+  igndbg << "Press Q" << std::endl;
+  this->verticalVel = this->kMaxVerticalVel;
+  this->KeyEvent(true, Qt::Key_Q);
 
-  // Cleanup
-  plugins.clear();
+  igndbg << "Press E" << std::endl;
+  this->verticalVel = -this->kMaxVerticalVel;
+  this->KeyEvent(true, Qt::Key_E);
 }
```

Comment:
Can we add a test for `Press S` and `Release E` for coverage?

Reply:
Good call: [b0a7347](https://github.com/gazebosim/gz-gui/pull/440/commits/b0a7347747aa882d4ed784162b33cbdddfb5f019)